src/generate/register.rs

```diff
@@ -357,17 +357,14 @@ pub fn fields(
                         if !sequential_indexes {
                             return Err(anyhow!("unsupported array indexes in {}", f.name));
                         }
-                        (first, None)
+                        first
                     } else {
-                        (0, de.dim_index.clone())
+                        0
                     }
                 } else {
-                    (0, None)
-                };
-                let suffixes: Vec<_> = match index {
-                    Some(ix) => ix,
-                    None => (0..de.dim).map(|i| (first + i).to_string()).collect(),
+                    0
                 };
+                let suffixes: Vec<_> = de.indexes().collect();
```

Comment:
are these changes intended to be included?

Reply:
Yes. It's just code cleanup.